PCAxis.Serializers/Parquet/ParquetBuilder.cs

```diff
@@ -0,0 +1,391 @@
+            { PXConstant.DATASYMBOL_3, string.IsNullOrEmpty(meta.DataSymbol3) ? PXConstant.DATASYMBOL_3_STRING : meta.DataSymbol3 },
+            { PXConstant.DATASYMBOL_4, string.IsNullOrEmpty(meta.DataSymbol4) ? PXConstant.DATASYMBOL_4_STRING : meta.DataSymbol4 },
+            { PXConstant.DATASYMBOL_5, string.IsNullOrEmpty(meta.DataSymbol5) ? PXConstant.DATASYMBOL_5_STRING : meta.DataSymbol5 },
+            { PXConstant.DATASYMBOL_6, string.IsNullOrEmpty(meta.DataSymbol6) ? PXConstant.DATASYMBOL_6_STRING : meta.DataSymbol6 },
+            { PXConstant.DATASYMBOL_7, string.IsNullOrEmpty(meta.DataSymbolSum) ? PXConstant.DATASYMBOL_7_STRING : meta.DataSymbolSum },
+            { PXConstant.DATASYMBOL_NIL, string.IsNullOrEmpty(meta.DataSymbolNIL) ? PXConstant.DATASYMBOL_NIL_STRING : meta.DataSymbolNIL }
+         };
+        }
+
+        /// <summary>
+        /// Populates the Parquet table based on the PXModel data and metadata.
+        /// </summary>
+        /// <returns>The populated Parquet table.</returns>
+        public Table PopulateTable()
+        {
+            int matrixSize = model.Data.MatrixColumnCount * model.Data.MatrixRowCount;
+            double[] data = new double[matrixSize];
+            int[] variableValueCounts = GetVariableValueCounts();
+            var indices = GenerateDataPointIndices(variableValueCounts);
+
+            for (int m = 0; m < matrixSize; m++)
+            {
+                data[m] = model.Data.ReadElement(m);
+            }
+
+            List<DataField> dataFields = CreateDataFields();
+
+            var table = new Table(dataFields.ToArray());
+
+            foreach (var index in indices)
+            {
+                var row = PopulateRow(index, dataFields, variableValueCounts, data);
+                table.Add(row);
+            }
+
+            return table;
+        }
+
+        /// <summary>
+        /// Creates the Parquet schema fields based on the PXModel metadata.
+        /// </summary>
+        /// <returns>The list of Parquet data fields.</returns>
+        private List<DataField> CreateDataFields()
+        {
+            List<DataField> dataFields = new List<DataField>();
+            int variableCount = model.Meta.Variables.Count;
+
+            for (int i = 0; i < variableCount; i++)
+            {
+                var variable = model.Meta.Variables[i];
+                if (variable.IsContentVariable && variable.Values.Count > 1)
+                {
+                    foreach (var value in variable.Values)
+                    {
+                        string columnName = $"{variable.Code}_{value.Code}";
+                        dataFields.Add(new DataField(columnName, typeof(double)));
+                        dataFields.Add(new DataField($"{columnName}_symbol", typeof(string))); // Add corresponding symbol column
+                    }
+                }
+                else
+                {
+                    if (variable.IsContentVariable)
```

Comment:
You will probably need to add a value and value_symbol field for PX files that have not defined the content variable.

Reply:
If I understand you correctly you are referring to PX files without an explicit content/measure? In that case in line 328 I create the content. The tests actually serializes all the PX files, without errors. There are also PX files there with no explicit content as well. If I misunderstood, would you mind providing me with a PX file that won't serialize correctly?